### PR TITLE
Social: Add Auto-Conversion feature UI toggles

### PR DESCRIPTION
**Pull request by gmjuhasz**
This change adds the UI toggles to the Social admin page and under Jetpack Settings, for our upcoming auto-conversion feature.

CleanShot 2023-08-21 at 11 38 34 png

CleanShot 2023-08-21 at 11 38 47 png

## Proposed changes:

* Added the needed initial state changes in b055a20ec69a7df13d0bc744ae6cfe602fbd1696
* Extended the Jetpack Social store in `publicize-components` to be able to persist the changes for the auto-conversion feature 1ee71f353ea856161905aa043f1b3561b76d2196
* Added the actual toggle to the Social Admin page 55534ce24785c571f6522c9f5ac1bfda2fd376d7
* Added the toggle under Jetpack Settings cf7eafe86e98006bf55e228ecb007512b7a253bc

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/orgs/Automattic/projects/733/views/2?pane=issue&itemId=36021083
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Open up the Jetpack Social admin page, and the Jetpack Settings page's sharing section
  * You should not see the auto-conversion toggle anywhere
* Add the `jetpack-social-image-auto-convert` blog sticker to your site
  * Now you should be able to see the toggles.
* Test the functionality of the toggles
  * Toggling the state on/off should be reflected on the UI, and should stay persisted after browser refresh
  * Change the state in the Social admin page, then refresh the page on the Jetpack Settings, it should get updated too. Do it the other way around.
* Check the help URL-s it should redirect you to the support page for the media restrictions.

### CODE REVIEW

File: projects/plugins/social/src/class-jetpack-social.php [hunk 1]
```diff
@@ -254,6 +255,10 @@ public function initial_state() {
 						'enabled'         => $sig_settings->is_enabled(),
 						'defaultTemplate' => $sig_settings->get_default_template(),
 					),
+					'autoConversionSettings'       => array(
+						'available'       => $auto_conversion_settings->is_available( 'image' ),
+						'auto-conversion' => $auto_conversion_settings->is_enabled( 'auto-conversion' ),
+					),
```
Comment (spsiddarthan) [inline projects/plugins/social/src/class-jetpack-social.php:261]: Is it more straightforward to use `auto-conversion` or `image` everywhere?

File: projects/packages/publicize/src/class-publicize-base.php [hunk 2]
```diff
@@ -1792,6 +1792,11 @@ public function has_social_image_generator_feature() {
 	 * @return bool
 	 */
 	public function has_social_auto_conversion_feature( $type ) {
+		// For now we only have the image conversion, and we do not have a flag for the master switch, so
+		// we use the image flag for that. TBD.
+		if ( $type === 'auto-conversion' ) {
```
Comment (spsiddarthan) [inline projects/packages/publicize/src/class-publicize-base.php]: Maybe we should just pass in `image` here and not `auto-conversion`? Or we could pass in `auto-conversion` and update the feature name to `social-auto-convert` and make a diff to update the feature's name on WPCOM as well.

File: projects/plugins/jetpack/class.jetpack-gutenberg.php [hunk 3]
```diff
@@ -733,6 +735,8 @@ public static function enqueue_block_editor_assets() {
 				'dismissedNotices'                => $publicize->get_dismissed_notices(),
 				'isInstagramConnectionSupported'  => $publicize->has_instagram_connection_feature(),
 				'isMastodonConnectionSupported'   => $publicize->has_mastodon_connection_feature(),
+				'isAutoConversionAvailable'       => $auto_conversion_settings->is_available( 'image' ),
+				'isAutoConversionEnabled'         => $auto_conversion_settings->is_enabled( 'auto-conversion' ),
```
Comment (spsiddarthan) [inline projects/plugins/jetpack/class.jetpack-gutenberg.php]: Same comment as in other places to use just one key, either `image` or `auto-conversion`

File: projects/plugins/jetpack/_inc/client/sharing/features/auto-conversion-section.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+			<AutoConversionToggle toggleClass="jp-settings-sharing__sig-toggle">
+				<div>
+					<div>
+						<Text>
+							<strong>{ __( 'Automatically convert images for compatibility', 'jetpack' ) }</strong>
+						</Text>
+					</div>
+					{ createInterpolateElement(
+						__(
+							'Social media platforms require different image file types, and sizes. Upload one image and it will be automatically converted to ensure maximum compatibility & quality across all your connected platforms. <link>Learn more about media requirements.</link>',
```
Comment (spsiddarthan) [inline projects/plugins/jetpack/_inc/client/sharing/features/auto-conversion-section.jsx]: `image file types, and sizes`, do we need the comma after types?

File: projects/plugins/jetpack/_inc/client/sharing/features/auto-conversion-section.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+							'jetpack'
+						),
+						{
+							link: (
+								<ExternalLink
+									href={ getRedirectUrl( 'jetpack-social-media-support-information' ) }
```
Comment (spsiddarthan) [inline projects/plugins/jetpack/_inc/client/sharing/features/auto-conversion-section.jsx:27]: I suppose we have created the redirect already here?

File: projects/plugins/social/src/js/components/auto-conversion-toggle/index.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+		<ToggleSection
+			title={ __( 'Automatically convert images for compatibility', 'jetpack-social' ) }
+			disabled={ isUpdating }
+			checked={ isEnabled }
+			onChange={ toggleStatus }
+		>
+			<Text className={ styles.text }>
+				{ createInterpolateElement(
+					__(
+						'Social media platforms require different image file types, and sizes. Upload one image and it will be automatically converted to ensure maximum compatibility & quality across all your connected platforms. <link>Learn more about media requirements.</link>',
```
Comment (spsiddarthan) [inline projects/plugins/social/src/js/components/auto-conversion-toggle/index.tsx]: Do we need the comma after types?

**Review by spsiddarthan — COMMENTED** [review]
This works well, @gmjuhasz! I have left some comments about the key we use `image` or `auto-coversion`. We could just use one key there. We can make a patch to update the feature name if it makes things simpler.

File: projects/plugins/jetpack/_inc/client/sharing/features/auto-conversion-section.jsx (hunk 5 above)
Comment (gmjuhasz) [reply projects/plugins/jetpack/_inc/client/sharing/features/auto-conversion-section.jsx:27]: Yes, it was already used with the notices in the editor

**Comment by spsiddarthan** [comment]
I think is very close, do you think we need to fix the way this looks? The external link icon is on the next line. 

Screenshot 2023-08-22 at 20 17 24

**Comment by pablinos** [comment]
> I think is very close, do you think we need to fix the way this looks? The external link icon is on the next line.

Yes that would be good. Maybe the whole link should go on a new line, and perhaps we could put in a non-breaking space or something so it wraps ok on smaller screens.

Should we default the option to enabled? I'm guessing that's a change in the API rather than this UI PR though.

**Review by spsiddarthan — APPROVED** [review]
LGTM

**Pull request merged**